### DISCUSSION TITLE
check if key exists

### DISCUSSION THREAD

**Discussion by electriquo** [TomWright/dasel#250] [Q&A] [answered]
Although I browsed [search](https://daseldocs.tomwright.me/selectors/search) and [search optional](https://daseldocs.tomwright.me/selectors/search-1) documentation, I did not find how to check whether a key exists (a functionality which is equivalent to [jq `has()`](https://stedolan.github.io/jq/manual/#has(key))).

The motivation:
1. Check whether the input has specific key. E.g. does `{}` has `.foo.bar`?
2. Read (select) a key only if it exists

**Comment by TomWright** [comment, marked as answer]
The `-` character when used in a dynamic/search selector can be used to check the key/property name in an array/object, however it doesn't work in this case.

Dasel returns an exit code of `1` if it failed to find the item though so I wonder if you could just run a basic selector and check the exit code?

```bash
~ [] $ echo '{"name":"Tom"}' | dasel -r json .name
"Tom"
~ [] $ echo $?
0
~ [] $ echo '{"name":"Tom"}' | dasel -r json .age
Error: could not query node: could not find value: no value found for selector: .age: map[name:Tom]
~ [] $ echo $?
1
```

**Reply by electriquo** [reply]
That is what I did and yet was wondering whether there is any better way.
Thanks.

**Reply by TomWright** [reply]
If there's a lot of interest I could add a command but seeing as it's already possible it's not a high priority
